src/vs/vscode.proposed.d.ts

```diff
@@ -1045,4 +1045,20 @@ declare module 'vscode' {
 	}
 
 	//#endregion
+
+	//#region scm, Rachel & Peng https://github.com/Microsoft/vscode/issues/53892
+
+	export namespace scm {
+		/**
+		 * The currently visible [source controls](#SourceControl).
+		 */
+		export const visibleSourceControls: SourceControl[];
+
+		/**
+		 * An [event](#Event) which fires when the visible [source controls](#SourceControl) have changed.
+		 */
+		export const onDidChangeVisibleSourceControls: Event<SourceControl[]>;
+	}
+
+	//#endregion
```

Comment:
We should actually expose readonly versions of `SourceControl` instances. We don't want other extensions to be able to modify SourceControl instances.